schemas/README.md

```diff
@@ -825,6 +1157,137 @@ Intensity of detections
 </tr>
 </table>
 
+## LinePrimitive
+
+A primitive representing a series of points connected by lines
+
+<table>
+  <tr>
+    <th>field</th>
+    <th>type</th>
+    <th>description</th>
+  </tr>
+<tr>
+<td><code>type</code></td>
+<td>
+
+[enum LineType](#enum-linetype)
+
+</td>
+<td>
+
+Drawing primitive to use for lines
+
+</td>
+</tr>
+<tr>
+<td><code>pose</code></td>
+<td>
+
+[Pose](#pose)
+
+</td>
+<td>
+
+Origin of lines relative to reference frame
+
+</td>
+</tr>
+<tr>
+<td><code>thickness</code></td>
+<td>
+
+float64
+
+</td>
+<td>
+
+Line thickness
+
+</td>
+</tr>
+<tr>
+<td><code>scale_invariant</code></td>
+<td>
+
+boolean
+
+</td>
+<td>
+
+Indicates whether `thickness` is a fixed size in screen pixels (true), or specified in world coordinates and scales with distance from the camera (false)
+
+</td>
+</tr>
+<tr>
+<td><code>points</code></td>
+<td>
+
+[Point3](#point3)[]
+
+</td>
+<td>
+
+Points along the line
+
+</td>
+</tr>
+<tr>
+<td><code>color</code></td>
+<td>
+
+[Color](#color)
+
+</td>
+<td>
+
+Solid color to use for the whole line. One of `color` or `colors` must be provided.
+
+</td>
+</tr>
+<tr>
+<td><code>colors</code></td>
+<td>
+
+[Color](#color)[]
+
+</td>
+<td>
+
+Per-point colors (if specified, must have the same length as `points`). One of `color` or `colors` must be provided.
```

Comment:
Seems like this complexity could be removed. If people want per-segment colors, they can already do that by adding multiple Lines to a SceneEntity.

Reply:
> If people want per-segment colors, they can already do that by adding multiple Lines to a SceneEntity.

That kind of works, except that multiple Lines in a SceneEntity can't be connected in a polygon or line strip — they would just be like a line list (which means sending extra copies of points, if they need to be the last point of one line & the first point of the next).

Reply:
Also, we would have to verify that each Line primitive has the same `pose` in order to render them efficiently in a batch. With this current design we are trying to avoid wasting CPU time on auto-batching, rather just allowing the user to provide data in batches if they need more efficient rendering. All the segments (with per-point colors) could be included in one batch with the current design.

Reply:
👍 yeah I did not fully grasp the rendering perf impacts here